projects/ngneat/error-tailor/src/lib/control-error.directive.spec.ts

```diff
@@ -315,4 +316,120 @@ describe('ControlErrorDirective', () => {
+    }
+
+    @Component({
+      template: `
+        <h1>{{ _text }}</h1>
+      `
+    })
+    class CustomControlErrorComponent implements ControlErrorComponent {
+      _text: string | null;
+      hide = true;
+
+      set customClass(className: string) {
+        this.host.nativeElement.classList.add(className);
+      }
+
+      set text(value: string | null) {
+        if (value !== this._text) {
+          this._text = value;
+          this.hide = !value;
+          this.cdr.markForCheck();
+        }
+      }
+      constructor(private cdr: ChangeDetectorRef, private host: ElementRef<HTMLElement>) {}
+    }
+
+    function getCustomErrorComponentFactory<C>(
+      component: Type<C>,
+      controlErrorComponent: Type<any> = null,
+      controlErrorComponentAnchorFn: (hostElem: Element, errorElem: Element) => () => void = null
+    ) {
+      return createComponentFactory({
+        component,
+        declarations: [CustomControlErrorComponent],
+        imports: [
+          FormsModule,
+          ReactiveFormsModule,
+          ErrorTailorModule.forRoot({
+            errors: {
+              useValue: {
+                required: () => 'required error'
+              }
+            },
+            controlErrorComponent: CustomControlErrorComponent,
+            controlErrorComponentAnchorFn: controlErrorComponentAnchorFn
+          })
+        ]
+      });
+    }
+
+    describe('CustomControlErrorComponent', () => {
+      let spectator: Spectator<CustomErrorFormGroupComponent>;
+      const createComponent = getCustomErrorComponentFactory(
+        CustomErrorFormGroupComponent,
+        CustomControlErrorComponent
+      );
+
+      beforeEach(() => (spectator = createComponent()));
+
+      it('should create custom error component', () => {
+        const input = spectator.query<HTMLInputElement>(byPlaceholder('Name'));
+
+        typeInElementAndFocusOut(spectator, '', input);
+
+        expect(spectator.query('h1')).toBeTruthy();
+        expect(spectator.query(byText('required error'))).toBeTruthy();
+      });
+    });
+
+    describe('ErrorComponentAnchorFnCallback', () => {
+      let anchorFnCalled = false;
+      let anchorFnDestroyCalled = false;
+
+      let spectator: Spectator<CustomErrorFormGroupComponent>;
+      const createComponent = getCustomErrorComponentFactory(
+        CustomErrorFormGroupComponent,
+        null,
+        (hostElem: Element, errorElem: Element) => {
+          anchorFnCalled = true;
+          expect(hostElem).toBeTruthy();
+          expect(errorElem).toBeTruthy();
+          return () => {
+            anchorFnDestroyCalled = true;
+          };
+        }
+      );
+
+      beforeEach(() => (spectator = createComponent()));
+
+      it('should call error component anchor callback', () => {
+        const input = spectator.query<HTMLInputElement>(byPlaceholder('Name'));
+
+        typeInElementAndFocusOut(spectator, '', input);
+
+        expect(anchorFnCalled).toBeTruthy();
```

Comment:
What about the destroy?

Reply:
I'm finding it hard to write a test for this for I couldn't find a way to destroy a component from a test. One can write a  test which relies on a previous test having destroyed the component and the destroyFn having been called. But this would mean that the tests are interdependent, which doesn't sound right. Any ideas will be appreciated.

Reply:
You can create a host component that renders a child component. https://github.com/ngneat/spectator#custom-host-component

Reply:
Thanks.

Reply:
I have added a test for destroy callback too. Using the `ngIf` technique that you had suggested, but incorporated the control variable in the parent form itself to control the field's rendering. So when the control variable, `showName` is set to false and the form is rendered, the erroneous field is removed which requires the attached error-control to be destroyed as well and this sets off the callback invocation. Pls have a look and let me know if there are any issues with this approach.